### PR TITLE
Fix devops artifact cannot be download due to auth checking

### LINKED ISSUE #3083: DevOps Project admin cannot download artifacts
**Describe the Bug**
DevOps Project admin cannot download artifacts.

image

**Versions Used**
KubeSphere: v3.0.0

/kind bug
/milestone 3.1.0
/assign @wansir

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind bug
/area devops

**What this PR does / why we need it**:
This PR can fix that artifacts from pipeline cannot be download.

**Which issue(s) this PR fixes**:
Fixes #3083

**Special notes for reviewers**:
I suggest that the front-end should replace API call from `/jenkins/{path:*}` to `/devops/{devops}/jenkins/{path:*}`

**Additional documentation, usage docs, etc.**:

### CODE REVIEW

File: pkg/kapis/devops/v1alpha2/register.go [hunk 1]
```diff
@@ -688,6 +688,9 @@ func AddJenkinsToContainer(webservice *restful.WebService, devopsClient devops.I
 		return err
 	}
 	parse.Path = strings.Trim(parse.Path, "/")
+	// this API does not belong any kind of auth scope, it should be removed in the future version
+	// see also pkg/apiserver/request/requestinfo.go
+	// please use /devops/{devops}/jenkins/{path:*} instead
 	webservice.Route(webservice.GET("/jenkins/{path:*}").
```
Comment (zryfish) [inline pkg/kapis/devops/v1alpha2/register.go:694]: please mark it as deprecated if we are going retire it

Comment (LinuxSuRen) [reply pkg/kapis/devops/v1alpha2/register.go:694]: The deprecated flag is ready. https://github.com/kubesphere/kubesphere/pull/3190/files#diff-0a2d700995b0afc6e606eae4e730b1445bf6c7b18f170c3862595b341bfd380aR693

/cc @shaowenchen 
/cc @zryfish

**Pull request merged**